packages/block-library/src/search/view.js

```diff
@@ -1,172 +1,75 @@
-	searchButton.removeAttribute( 'aria-controls' ); // Note: Seemingly not reflected with searchButton.ariaControls.
-	toggleAriaLabel( searchButton );
-	block.classList.remove( hiddenClass );
-
-	searchField.focus(); // Note that Chrome seems to do this automatically.
-
-	// The following two must be inverse of what is done in collapseExpandedSearchBlock().
-	document.addEventListener( 'keydown', handleKeydownEvent, {
-		passive: true,
-	} );
-	document.addEventListener( 'keyup', handleKeyupEvent, {
-		passive: true,
-	} );
-
-	expandedSearchBlock = block;
-}
-
 /**
- * Collapses the expanded search block.
- *
- * Inverse of what is done in expandSearchBlock().
+ * WordPress dependencies
  */
-function collapseExpandedSearchBlock() {
-	if ( ! expandedSearchBlock ) {
-		throw new Error( 'Expected expandedSearchBlock to be defined.' );
-	}
-	const block = expandedSearchBlock;
-	const searchField = getSearchInput( block );
-	const searchButton = getSearchButton( block );
-
-	searchButton.type = 'button';
-	searchField.ariaHidden = 'true';
-	searchField.tabIndex = -1;
-	searchButton.ariaExpanded = 'false';
-	searchButton.setAttribute( 'aria-controls', searchField.id ); // Note: Seemingly not reflected with searchButton.ariaControls.
-	toggleAriaLabel( searchButton );
-	block.classList.add( hiddenClass );
-
-	// The following two must be inverse of what is done in expandSearchBlock().
-	document.removeEventListener( 'keydown', handleKeydownEvent, {
-		passive: true,
-	} );
-	document.removeEventListener( 'keyup', handleKeyupEvent, {
-		passive: true,
-	} );
-
-	expandedSearchBlock = null;
-}
-
-// Listen for click events anywhere on the document so this script can be loaded asynchronously in the head.
-document.addEventListener(
-	'click',
-	( event ) => {
-		// Get the ancestor expandable Search block of the clicked element.
-		const block = event.target.closest(
-			'.wp-block-search__button-behavior-expand'
-		);
-
-		/*
-		 * If there is already an expanded search block and either the current click was not for a Search block or it was
-		 * for another block, then collapse the currently-expanded block.
-		 */
-		if ( expandedSearchBlock && block !== expandedSearchBlock ) {
-			collapseExpandedSearchBlock();
-		}
-
-		// If the click was on or inside a collapsed Search block, expand it.
-		if (
-			block instanceof HTMLFormElement &&
-			block.classList.contains( hiddenClass )
-		) {
-			expandSearchBlock( block );
-		}
+import { store as wpStore } from '@wordpress/interactivity';
+
+wpStore( {
+	selectors: {
+		core: {
+			search: {
+				ariaLabel: ( { context } ) => {
+					const { ariaLabelCollapsed, ariaLabelExpanded } =
+						context.core.search;
+					return context.core.search.isSearchInputVisible
+						? ariaLabelExpanded
+						: ariaLabelCollapsed;
+				},
+				ariaControls: ( { context } ) => {
+					return context.core.search.isSearchInputVisible
+						? undefined
+						: context.core.search.inputId;
```

Comment:
Using `null` feels a bit more explicit.

```suggestion
					return context.core.search.isSearchInputVisible
						? null
						: context.core.search.inputId;
```